src/new-components/base-field/base-field.tsx

```diff
@@ -6,28 +6,107 @@ import styles from './base-field.module.css'
+            {...props}
+        >
+            <path
+                fillRule="evenodd"
+                clipRule="evenodd"
+                d="M8 2.5C4.96243 2.5 2.5 4.96243 2.5 8C2.5 11.0376 4.96243 13.5 8 13.5C11.0376 13.5 13.5 11.0376 13.5 8C13.5 4.96243 11.0376 2.5 8 2.5ZM1.5 8C1.5 4.41015 4.41015 1.5 8 1.5C11.5899 1.5 14.5 4.41015 14.5 8C14.5 11.5899 11.5899 14.5 8 14.5C4.41015 14.5 1.5 11.5899 1.5 8ZM8.66667 10.3333C8.66667 10.7015 8.36819 11 8 11C7.63181 11 7.33333 10.7015 7.33333 10.3333C7.33333 9.96514 7.63181 9.66667 8 9.66667C8.36819 9.66667 8.66667 9.96514 8.66667 10.3333ZM8.65766 5.65766C8.65766 5.29445 8.36322 5 8 5C7.99087 5.00008 7.98631 5.00013 7.98175 5.00025C7.97719 5.00038 7.97263 5.00059 7.96352 5.00101C7.60086 5.02116 7.3232 5.33149 7.34335 5.69415L7.50077 8.52774C7.53575 9.15742 8.46425 9.15742 8.49923 8.52774L8.65665 5.69415C8.65707 5.68503 8.65728 5.68047 8.65741 5.67591C8.65754 5.67135 8.65758 5.66679 8.65766 5.65766Z"
+                fill="currentColor"
+            />
+        </svg>
+    )
+}
+
+function SuccessMessageIcon(props: React.SVGProps<SVGSVGElement>) {
+    return (
+        <svg
+            width="16"
+            height="16"
+            viewBox="0 0 16 16"
+            fill="none"
+            xmlns="http://www.w3.org/2000/svg"
+            {...props}
+        >
+            <path
+                d="M5.63807 7.86191C5.37772 7.60156 4.95561 7.60156 4.69526 7.86191C4.43491 8.12226 4.43491 8.54437 4.69526 8.80472L6.36193 10.4714C6.62228 10.7317 7.04439 10.7317 7.30474 10.4714L11.3047 6.47138C11.5651 6.21103 11.5651 5.78892 11.3047 5.52858C11.0444 5.26823 10.6223 5.26823 10.3619 5.52858L6.83333 9.05717L5.63807 7.86191Z"
+                fill="currentColor"
+            />
+        </svg>
+    )
+}
+
+type FieldTone = 'neutral' | 'success' | 'error' | 'loading'
+
+type FieldMessageProps = FieldHintProps & {
+    tone: FieldTone
+}
+
+function FieldMessage({ id, children, tone }: FieldMessageProps) {
+    const textTone = tone === 'error' ? 'danger' : tone === 'success' ? 'positive' : 'normal'
+    return (
+        <Text as="p" tone={textTone} size="copy" id={id}>
+            {tone && tone !== 'neutral' ? (
+                <Box
+                    as="span"
+                    marginRight="xsmall"
+                    display="inlineFlex"
+                    className={styles.messageIcon}
+                >
+                    {tone === 'error' ? (
+                        <ErrorMessageIcon aria-hidden />
+                    ) : tone === 'success' ? (
+                        <SuccessMessageIcon aria-hidden />
+                    ) : tone === 'loading' ? (
+                        <Spinner size={16} />
+                    ) : null}
+                </Box>
+            ) : null}
+            {children}
+        </Text>
+    )
+}
+
 //
 // BaseField
 //
 
+type ChildrenRenderProps = {
+    id: string
+    'aria-describedby'?: string
+    'aria-errormessage'?: string
```

Comment:
Maybe it's not an issue since the markup here is similar to mdn's example, but the error message itself isn't read when the field receives focus. It only gets read when you traverse to the message itself, but even then there's no indication that it's linked to the text field:

![image](https://user-images.githubusercontent.com/8531248/171623378-39af93ae-d3d9-45fd-92c6-cc5cccf981a1.png)
![image](https://user-images.githubusercontent.com/8531248/171623369-777e104f-77a2-46ce-84c6-7503f48f367c.png)

Reply:
Apparently, support for this is not great on VoiceOver (or in general) 😕

https://a11ysupport.io/tech/aria/aria-errormessage_attribute

cc @craigcarlyle what do you think we could do? Maybe we make it part of the description all the time? I tested this, and VoiceOver still announces that the field is marked as invalid. But it does not announce the error text. So having it in the description would have it read out loud.

Reply:
@gnapse Since `aria-errormessage` came out in WAI-ARIA 1.1 in _2017_ and is _still_ not supported across assistive software I think we have no choice but to use `aria-describedby`. 

I think the suggestion [here](https://www.davidmacd.com/blog/test-aria-describedby-errormessage-aria-live.html) is the way to go:
![Screen Shot 2022-06-06 at 3 15 04 PM](https://user-images.githubusercontent.com/1305500/172258250-da744f35-49f9-44fc-9f61-7574bb4f59d7.png)

Reply:
Done in the last commit.